engine/sound/src/devices/device_openal.cpp

```diff
@@ -120,44 +128,12 @@ namespace dmDeviceOpenAL
         alcProcessContext(openal->m_Context);
 
         alSourceStop(openal->m_Source);
-
-        int iter = 0;
-        while (openal->m_Buffers.Size() != openal->m_Buffers.Capacity())
-        {
-            ALint num_queued = 0;
-            alcProcessContext(openal->m_Context);
-            alGetSourcei(openal->m_Source, AL_BUFFERS_QUEUED, &num_queued);
-            CheckAndPrintError();
-
-            while (num_queued > 0) {
-                ALuint buffer;
-                alSourceUnqueueBuffers(openal->m_Source, 1, &buffer);
-                if (CheckAndPrintError() != AL_NO_ERROR)
-                {
-                    dmLogError("Still buffers in OpenAL. Bailing.");
-                    break;
-                }
-
-                openal->m_Buffers.Push(buffer);
-
-                alGetSourcei(openal->m_Source, AL_BUFFERS_QUEUED, &num_queued);
-            }
-
-            if ((iter + 1) % 10 == 0) {
-                dmLogInfo("Waiting for OpenAL device to complete");
-            }
-            ++iter;
-            dmTime::Sleep(10 * 1000);
-
-            if (iter > 1000) {
-                dmLogError("Still buffers in OpenAL. Bailing.");
-                break;
-            }
-        }
+        CheckAndPrintError();
 
         alDeleteSources(1, &openal->m_Source);
+        CheckAndPrintError();
 
-        alDeleteBuffers(openal->m_Buffers.Size(), openal->m_Buffers.Begin());
+        alDeleteBuffers(openal->m_AllBuffers.Size(), openal->m_AllBuffers.Begin());
```

Comment:
We've had our old iterative solution since 2014 (8d01fa7c839), but I'm not sure why it was written that way to begin with.

Reply:
I remember some issues on iOS. I think new solution should be carefully tested on iOS (esp. with going background/foreground, closing app, and restart)

Reply:
Yeah, this is my main concern as well. But the old solution came in place 10 years ago. A lot has happened since.

Reply:
I do like that the new solution is so clean and without that loop though!